### PR TITLE
feat: autocommit sample

### PR DESCRIPTION
**Pull request by AlisskaPie**
These changes have been moved from python-spanner-django [PR](https://github.com/googleapis/python-spanner-django/pull/552).

### CODE REVIEW

**Comment by IlyaFaer** [comment]
I see an error `ModuleNotFoundError: No module named sqlparse` here. We already have two PRs with adding this dependency:
https://github.com/googleapis/python-spanner/pull/171/files
https://github.com/googleapis/python-spanner/pull/168/files

**Comment by c24t** [comment]
Looks like I gave you some bad info in https://github.com/googleapis/python-spanner-django/pull/552#discussion_r518406244, making `samples` a package causes import errors for other sample modules:

```
________________ ERROR collecting samples/samples/backup_sample_test.py _______________
ImportError while importing test module 'src/python-spanner/samples/samples/backup_sample_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.pyenv/versions/3.8.2/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
backup_sample_test.py:21: in <module>
    import backup_sample
E   ModuleNotFoundError: No module named 'backup_sample'
__________________ ERROR collecting samples/samples/quickstart_test.py _______________
ImportError while importing test module 'src/python-spanner/samples/samples/quickstart_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.pyenv/versions/3.8.2/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
quickstart_test.py:21: in <module>
    import quickstart
E   ModuleNotFoundError: No module named 'quickstart'
___________________ ERROR collecting samples/samples/snippets_test.py _______________
ImportError while importing test module 'src/python-spanner/samples/samples/snippets_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.pyenv/versions/3.8.2/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
snippets_test.py:21: in <module>
    import snippets
E   ModuleNotFoundError: No module named 'snippets'
```

Should be fixed now.

**Comment by AlisskaPie** [comment]
> Looks like I gave you some bad info in [googleapis/python-spanner-django#552 (comment)](https://github.com/googleapis/python-spanner-django/pull/552#discussion_r518406244), making `samples` a package causes import errors for other sample modules:
...
> Should be fixed now.

Thanks for commiting code! Errors in importing other sample modules have disappeared.

File: samples/samples/autocommit.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """Enables autocommit mode."""
+    # [START spanner_enable_autocommit_mode]
+    from google.cloud.spanner_dbapi.checksum import ResultsChecksum
+
+    connection = connect(instance_id, database_id)
+    connection.autocommit = True
+    print("Autocommit mode is enabled.")
+
+    cursor = connection.cursor()
+    cursor._checksum = ResultsChecksum()
```
Comment (larkee) [inline samples/samples/autocommit.py:22]: Why does the user need to initialise a private variable? Can this not be initialised as part of the cursor factory?

**Review by larkee — COMMENTED** [review]
Mostly LGTM, just the one concern

File: samples/samples/autocommit.py (hunk 1 above)
Comment (AlisskaPie) [reply samples/samples/autocommit.py]: Thank you for the review! And sorry for the delayed answer.
I've made an update and placed this part in the test.

Comment (larkee) [reply samples/samples/autocommit.py]: This is not quite what I meant. The sample now looks as I would expect but I'm unclear as to why you still need to set `cursor._checksum` in the test. Is there a reason that [`_checksum` defaults to `None`](https://github.com/googleapis/python-spanner/blob/master/google/cloud/spanner_dbapi/cursor.py#L61)? Could it be updated to `ResultsChecksum()` instead?

File: samples/samples/autocommit_test.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Google LLC
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bsd
+
+import uuid
+
+from google.cloud import spanner
+import pytest
```
Comment (c24t) [inline samples/samples/autocommit_test.py:10]:
```suggestion
import mock
import pytest
from google.cloud import spanner
from google.cloud.spanner_dbapi import connect
```

Move imports up.

File: samples/samples/autocommit_test.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+INSTANCE_ID = unique_instance_id()
+DATABASE_ID = unique_database_id()
+
+
+@pytest.fixture(scope="module")
+def spanner_instance():
+    spanner_client = spanner.Client()
+    config_name = f"{spanner_client.project_name}/instanceConfigs/regional-us-central1"
```
Comment (c24t) [inline samples/samples/autocommit_test.py:32]:
```suggestion
    config_name = (
        f"{spanner_client.project_name}/instanceConfigs/regional-us-central1"
    )
```

Format with black.

File: samples/samples/autocommit_test.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    db.create()
+    yield db
+    db.drop()
+
+
+def test_enable_autocommit_mode(capsys, database):
+    import mock
+    from google.cloud.spanner_dbapi.checksum import ResultsChecksum
+    from google.cloud.spanner_dbapi import connect
+
```
Comment (c24t) [inline samples/samples/autocommit_test.py:54]:
```suggestion
```

Move imports up, lose ResultsChecksum.

File: samples/samples/autocommit_test.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    connection = connect(INSTANCE_ID, DATABASE_ID)
+    cursor = connection.cursor()
+    cursor._checksum = ResultsChecksum()
```
Comment (c24t) [inline samples/samples/autocommit_test.py:57]:
```suggestion
```

**Review by c24t — COMMENTED** [review]
Re: https://github.com/googleapis/python-spanner/pull/172#discussion_r535791975, it looks like the tests pass without setting `cursor._checksum` and I see you didn't need this line in https://github.com/googleapis/python-spanner-django/pull/552. Any reason not to remove it here?

The other suggested changes are just formatting and liberating some imports.

As in https://github.com/googleapis/python-spanner-django/pull/552#pullrequestreview-524719695, I think the sample would be more useful if it showed the different behavior with autocommit on/off, but this sample looks fine to me otherwise. LGTM with my suggested changes.

**Comment by IlyaFaer** [comment]
@googlebot I consent.

**Comment by IlyaFaer** [comment]
One system test `create_instance` failed with `429 Project 1065521786570 cannot add 1 nodes in region us-central1`. Doesn't seem to be related.

**Comment by IlyaFaer** [comment]
Hm-m, it looks like this `ResourceExhausted` error became permanent. Maybe it'll be better to move DB API system tests to the original `test_system.py` file and use one instance for all of the system tests?

**Comment by larkee** [comment]
> Hm-m, it looks like this ResourceExhausted error became permanent. Maybe it'll be better to move DB API system tests to the original test_system.py file and use one instance for all of the system tests?

The issue is test instances failing to be deleted for some runs. Manually deleting instances from the test project can resolve this temporarily. For a more long term solution, see #190

**Review by larkee — APPROVED** [review]

**Review by c24t — APPROVED** [review]
LGTM, let's get a green system test build.

**Pull request merged**